ecommerce/extensions/checkout/mixins.py

```
@@ -59,6 +95,13 @@ def handle_successful_order(self, order):
             user_id=order.user.id
         )
 
-        post_checkout.send_robust(sender=self, order=order)
+        if waffle.switch_is_active('async_order_fulfillment'):
+            # Always commit transactions before sending tasks depending on state from the current transaction!
+            # There's potential for a race condition here if the task starts executing before the active
+            # transaction has been committed; the necessary order doesn't exist in the database yet.
+            # See http://celery.readthedocs.org/en/latest/userguide/tasks.html#database-transactions.
+            fulfill_order.delay(order.number)
+        else:
+            post_checkout.send_robust(sender=self, order=order)
```

@jimabramson good question. It may simply have been an oversight. `send_robust()` guarantees that all receivers are notified of the emitted signal, preventing program flow from being interrupted by an exception in a receiver. However, it also swallows these exceptions, and we don't inspect the values returned by the receivers to check for errors.

@clintonb, you originally suggested that we use `send_robust()` here. Would you agree with reverting to `send()`? Being noisy about errors will make debugging easier for us. Plus, we only have a single receiver attached to the `post_checkout` signal at the moment, so I don't really see the benefit of `send_robust()`.